dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java

```diff
@@ -454,27 +454,25 @@ public Entry createAndSetDiscourseTopic(Long id) throws CustomWebApplicationExce
 
         // Create title and link to entry
 
-        String entryLink = "https://dockstore.org/";
-        String title = "";
-        if (hostName.contains("staging")) {
-            entryLink = "https://staging.dockstore.org/";
-            title = "Staging ";
-        }
+        final String entryPath;
+        final String sitePath;
         if (entry instanceof BioWorkflow) {
-            title += ((BioWorkflow)(entry)).getWorkflowPath();
-            entryLink += "workflows/";
+            entryPath = ((BioWorkflow)entry).getWorkflowPath();
+            sitePath = "workflows/" + entryPath;
         } else if (entry instanceof Service) {
-            title += ((Service)(entry)).getWorkflowPath();
-            entryLink += "services/";
+            entryPath = ((Service)entry).getWorkflowPath();
+            sitePath = "services/" + entryPath;
         } else if (entry instanceof AppTool) {
-            title += ((AppTool)(entry)).getWorkflowPath();
-            entryLink += "tools/";
+            entryPath = ((AppTool)entry).getWorkflowPath();
+            sitePath = "tools/" + entryPath;
         } else {
-            title += ((Tool)(entry)).getToolPath();
-            entryLink += "tools/";
+            entryPath = ((Tool)entry).getToolPath();
+            sitePath = "tools/" + entryPath;
         }
 
-        entryLink += title;
+        final boolean isStaging = hostName.contains("staging");
+        final String title = (isStaging ? "Staging " : "") + entryPath;
+        final String entryLink = (isStaging ? "https://staging.dockstore.org/" : "https://dockstore.org/") + sitePath;
```

Comment:
It was like this, but there's a more [generic way](https://github.com/dockstore/dockstore/blob/c4f86a7aed29ed226799ed774256a567a7568777/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java#L213) to determine the base url. And you could extract the title from the subdomain.

Probably over-architecting this for such a simple change; just giving another way to handle Denis' comment, which I think is worth doing.

Reply:
If we're not wedded to adding a single word, we might as well append the base url in the non-prod cases, it conveys more information.

Reply:
Both are fine with me, just may be nice to do this in some way while we're rummaging around in here